src/components/CentralContainer/Sidebar/SidebarTabs/SearchTabPanel/index.tsx

```diff
@@ -0,0 +1,119 @@
+import React, {
+    useContext,
+    useRef,
+    useState,
+} from "react";
+
+import {
+    AccordionGroup,
+    IconButton,
+    LinearProgress,
+    Textarea,
+    ToggleButtonGroup,
+} from "@mui/joy";
+
+import UnfoldLessIcon from "@mui/icons-material/UnfoldLess";
+import UnfoldMoreIcon from "@mui/icons-material/UnfoldMore";
+
+import {StateContext} from "../../../../../contexts/StateContextProvider";
+import {
+    TAB_DISPLAY_NAMES,
+    TAB_NAME,
+} from "../../../../../typings/tab";
+import CustomTabPanel from "../CustomTabPanel";
+import TitleButton from "../TitleButton";
+import ResultsGroup from "./ResultsGroup";
+
+
+enum SEARCH_OPTION {
+    IS_CASE_SENSITIVE = "isCaseSensitive",
+    IS_REGEX = "isRegex"
+}
+
+/**
+ * Displays a panel for submitting search queries and viewing query results.
+ *
+ * @return
+ */
+const SearchTabPanel = () => {
+    const [isAllExpanded, setIsAllExpanded] = useState<boolean>(true);
+    const [searchOptions, setSearchOptions] = useState<SEARCH_OPTION[]>([]);
+    const searchTextRef = useRef<HTMLTextAreaElement>(null);
+    const {queryResults, startQuery} = useContext(StateContext);
```

Comment:
Using contexts are considered imports. Let's move this to the very first line of the function.